config/src/main/java/org/springframework/security/config/annotation/web/configuration/HttpSecurityConfiguration.java

```diff
@@ -131,8 +133,13 @@ HttpSecurity httpSecurity() throws Exception {
 	}
 
 	private void applyCorsIfAvailable(HttpSecurity http) throws Exception {
-		String[] beanNames = this.context.getBeanNamesForType(CorsConfigurationSource.class);
-		if (beanNames.length == 1) {
+		Map<String, CorsConfigurationSource> corsConfigurationSources = this.context
+			.getBeansOfType(CorsConfigurationSource.class);
+
+		boolean hasUrlBasedCorsConfigurationSource = corsConfigurationSources.values()
```

Comment:
Would you please adopt another strategy instead of [using Stream](https://github.com/spring-projects/spring-security/issues/9642)?

Reply:
I have refactored the code to avoid using Streams. PTAL. Thanks.